app/src/main/kotlin/org/eclipse/kuksa/companion/feature/connection/view/AdaptiveConnectionStatusView.kt

```diff
@@ -44,12 +49,29 @@ fun AdaptiveConnectionStatusView(
     }
 }
 
+class ConnectionStatusView private constructor() {
+    companion object {
+        fun calculatePaddingValues(
+            viewModel: ConnectionStatusViewModel,
+            windowSizeClass: WindowSizeClass,
+        ): PaddingValues {
+            if (viewModel.connectionState == ConnectionState.CONNECTED) return PaddingValues(0.dp)
```

Comment:
I agree, this method is a bit unlucky. The naming "calculate" does not really fit, because it is not calculating anything and the viewModel feels off here... 

How about a AdaptiveStatusViewPadding(Values) class implementing the PaddingValues interface but taking the windowSizeClass into account? Maybe even split it into two classes like AdaptivePaddingValues and then the subclasses. 

You would have something like:
`AdaptiveStatusViewPadding(windowSizeClass)`
```
val padding = AdaptiveNavigationViewPadding(windowSizeClass)
padding.horizontal(offset = 12.dp)
```

For the viewModel I would recommend moving it out of here to the caller site.